src/04-extensions.js

```diff
@@ -0,0 +1,212 @@
+one of the ticket "extras", provide information about what someone would be able to do based on purchasing 
+that ticket. The format is up to you.
+*/
+
+function ticketExtraDetails(extras) {
+  //edge case - if the extra is not movie education or terrace return error message
+  const validExtras = ["movie", "education", "terrace"];
+
+  if (!Array.isArray(extras) && !validExtras.includes(extras.toLowerCase())) {
+    return "must be a valid extra: 'movie', 'education', or 'terrace'";
+  } else {
+    extras = extras.toLowerCase();
+  }
+
+  //if argument is not an array and data type is string
+  if (!Array.isArray(extras) && typeof extras === "string") {
+    if (extras === "education") return "If you purchase this ticket extra you will gain access to 2 more rooms and see 2 more dinosaurs!";
+    if (extras === "terrace") return "If you purchase this ticket extra you will gain access to 1 more room!";
+    if (extras === "movie") return "If you purchase this ticket extra you will gain access to 3 more rooms and see 2 more dinosaurs and see 1 jurassic park movie!";
+  }
+
+  //purpose of creating object is to create reusable code for if we ever want to add extras to extras
+  const extrasObj = {
+    //ordered list represents access to x more rooms, see x more dinos, and see 1 jurassic park movie
+    education: [2, 2, 0],
+    terrace: [1, 0, 0],
+    movie: [3, 2, 1],
+  };
+
+  //iterate through our array of strings called extras with reduce method
+  const result = extras.reduce(
+    (acc, cVal) => {
+      //if the current element is education
+      if (cVal === "education") {
+        //iterate through accumulator
+        acc.map((el, index) => {
+          //add and then re-assign element to evaluated result of adding extrasObject at education at index
+          acc[index] += extrasObj["education"][index];
+          return el;
+        });
+      } else if (cVal === "terrace") {
+        //iterate through accumulator
+        acc.map((el, index) => {
+          //add and then re-assign element to evaluated result of adding extrasObject at education at index
+          acc[index] += extrasObj["terrace"][index];
+          return el;
+        });
+      } else if (cVal === "movie") {
+        //iterate through accumulator
+        acc.map((el, index) => {
+          //add and then re-assign element to evaluated result of adding extrasObject at education at index
+          acc[index] += extrasObj["movie"][index];
+          return el;
+        });
+      }
+      return acc;
+      //get array there and add manually each index to empty array
+    },
+    [0, 0, 0]
+  );
+  //if last index is 0 ignore that index
+  if (result[result.length - 1] === 0) return `If you purchase this ticket extra you will gain access to ${result[0]} more rooms and see ${result[1]} more dinosaurs!`;
+  return `If you purchase this ticket extra you will gain access to ${result[0]} more rooms, see ${result[1]} more dinosaurs and see 1 jurassic park movie!`;
```

Comment:
This ends up being pretty specific for this dataset. What if the dataset changed so that there were `0` dinosaurs in any of the "extra" rooms?